pkg/docker/parseimage_test.go

```diff
@@ -35,38 +35,67 @@ func TestParseImageIDString(t *testing.T) {
 		shouldPass  bool
 	}{
 		{
-			description: "valid image name",
+			description: "valid format",
+			prefix:      "docker-pullable://",
+			name:        "abc",
+			sha:         "cb4983d8399a59bb5ee6e68b6177d878966a8fe41abe18a45c3b1d8809f1d043",
+			shouldPass:  true,
+		},
+		{
+			description: "valid format with 2 directories",
 			prefix:      "docker-pullable://",
 			name:        "abc/def",
 			sha:         "cb4983d8399a59bb5ee6e68b6177d878966a8fe41abe18a45c3b1d8809f1d043",
 			shouldPass:  true,
 		},
 		{
-			description: "no prefix",
+			description: "valid format with 3 directories",
+			prefix:      "docker-pullable://",
+			name:        "abc/def/ghi",
+			sha:         "cb4983d8399a59bb5ee6e68b6177d878966a8fe41abe18a45c3b1d8809f1d043",
+			shouldPass:  true,
+		},
+		{
+			description: "valid format with private registry",
+			prefix:      "docker-pullable://",
+			name:        "docker-registry.default.svc:5000/def/ghi",
+			sha:         "cb4983d8399a59bb5ee6e68b6177d878966a8fe41abe18a45c3b1d8809f1d043",
+			shouldPass:  true,
+		},
+		{
+			description: "missing prefix",
 			prefix:      "",
 			name:        "abc/def",
 			sha:         "cb4983d8399a59bb5ee6e68b6177d878966a8fe41abe18a45c3b1d8809f1d043",
-			shouldPass:  true,
+			shouldPass:  false,
```

Comment:
I believe this change is wrong.  This test is there specifically to test openshift's images in etcd.  They won't have a docker-pullable prefix.

Reply:
Fixed the issue, can you please review it again.